"npm run package" returns "npm ERR!" with Object.values & Object.entries

Hi Irvin. Glad I came across your package as callback hell was really slowing down my current lambda project.

I receive the following error when attempting "npm run package" on a script which utilizes Object.values || Object.entries. The entire script + async / await appear to work fine when uploaded to AWS without either Object methods.
Terminal output:

>npm run package

> es2017-lambda-boilerplate@0.3.1 package /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> npm run build && npm run package:pack


> es2017-lambda-boilerplate@0.3.1 build /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> npm run build:init && npm run build:js && npm run build:install


> es2017-lambda-boilerplate@0.3.1 build:init /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> rm -rf dist && mkdir dist


> es2017-lambda-boilerplate@0.3.1 build:js /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> cd src && babel . -d ../dist

>index.js -> ../dist/index.js
TypeError: main.js: Property body[1] of BlockStatement expected node to be of a type ["Statement"] but instead got "FunctionExpression"
    at validate (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/definitions/in>dex.js:109:13)
    at validator (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/definitions/in>dex.js:57:7)
    at Object.validate (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/definitions/in>dex.js:190:10)
    at validate (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/index.>js:505:9)
    at Object.builder (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/index.>js:466:7)
    at NodePath.replaceExpressionWithStatements (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-traverse/lib/path/replac>ement.js:206:54)
    at NodePath.insertBefore (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-traverse/lib/path/modifi>cation.js:52:10)
    at NodePath.unshiftContainer (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-traverse/lib/path/modifi>cation.js:239:15)
    at File.addHelper (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-core/lib/transformation/>file/index.js:388:17)
    at PluginPass.addHelper (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-core/lib/transformation/>plugin-pass.js:46:42)
>npm ERR! code ELIFECYCLE
>npm ERR! errno 1
>npm ERR! es2017-lambda-boilerplate@0.3.1 build:js: 'cd src && babel . -d ../dist'
>npm ERR! Exit status 1
>npm ERR!
>npm ERR! Failed at the es2017-lambda-boilerplate@0.3.1 build:js script.
>npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

>npm ERR! A complete log of this run can be found in:
>npm ERR!     /Users/arg/.npm/_logs/2018-01-27T07_33_28_532Z-debug.log
>npm ERR! code ELIFECYCLE
>npm ERR! errno 1
>npm ERR! es2017-lambda-boilerplate@0.3.1 build: 'npm run build:init && npm run build:js && npm run build:install'
>npm ERR! Exit status 1
>npm ERR!
>npm ERR! Failed at the es2017-lambda-boilerplate@0.3.1 build script.
>npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

>npm ERR! A complete log of this run can be found in:
>npm ERR!     /Users/arg/.npm/_logs/2018-01-27T07_33_28_548Z-debug.log
>npm ERR! code ELIFECYCLE
>npm ERR! errno 1
>npm ERR! es2017-lambda-boilerplate@0.3.1 package: 'npm run build && npm run package:pack'
>npm ERR! Exit status 1
>npm ERR!
>npm ERR! Failed at the es2017-lambda-boilerplate@0.3.1 package script.
>npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

>npm ERR! A complete log of this run can be found in:
>npm ERR!     /Users/arg/.npm/_logs/2018-01-27T07_33_28_565Z-debug.log

[I've added your email to my private repo if you need to checkout the source file](https://bitbucket.org/avolition/cryptolambda/src/31eece9d23c807c06135b50884c89b3f5d4db110/Binance_CLI/src/main.js?fileviewer=file-view-default).

--- a/.babelrc
+++ b/.babelrc
@@ -19,6 +19,7 @@
                 "commentEnd": "\n "
             }
         ],
+        "transform-es2015-modules-commonjs",
         "transform-es2017-object-entries",
         "transform-object-rest-spread",
         "transform-es2015-destructuring",

--- a/package.json
+++ b/package.json
@@ -14,6 +14,7 @@
         "babel-plugin-add-header-comment": "^1.0.3",
         "babel-plugin-array-includes": "^2.0.3",
         "babel-plugin-transform-es2015-destructuring": "^6.23.0",
+        "babel-plugin-transform-es2015-modules-commonjs": "^6.26.0",
         "babel-plugin-transform-es2017-object-entries": "0.0.4",
         "babel-plugin-transform-object-rest-spread": "^6.26.0",
         "babel-preset-env": "^1.6.1",

--- a/src/main.js
+++ b/src/main.js
@@ -21,6 +21,8 @@ export default (event, context) => {
             return restApiExample(event.data, context);
         case 'aws-sdk-example':
             return awsSdkExample(event.data, context);
+        case 'integration-test-example':
+            return integrationTestExample(event.data, context);
     }
 };
 
@@ -43,6 +45,36 @@ const restApiExample = async (data, context) => {
 const awsSdkExample = async (data, context) => {
     const EC2 = new AWS.EC2();
 
+    // return [
+    //     AWS.config.accessKeyId,
+    //     process.env.AWS_ACCESS_KEY_ID,
+    //     AWS.config.secretAccessKey,
+    //     process.env.AWS_SECRET_ACCESS_KEY,
+    // ];
+
     // Sample EC2 call.
     return await EC2.describeInstances().promise();
+};
+
+// Sample function used for integration testing against each ES2016/7 feature
+// purportedly supported by the boilerplate.
+const integrationTestExample = (data, context) => {
+    let value;
+
+    switch (data.function) {
+        case 'Object.values':
+            return Object.values(data.value);
+        case 'Object.entries':
+            return Object.entries(data.value);
+        case 'Array.prototype.includes':
+            return data.value.includes(data.value2);
+        case 'Exponentiation':
+            return data.value ** data.value2;
+        case 'Object spread properties':
+            const { value } = data;
+            return value;
+        case 'Object rest properties':
+            const { child, ...rest } = data.value;
+            return rest;
+    }
 };

--- a/test/integration.test.js
+++ b/test/integration.test.js
@@ -1,0 +1,117 @@
+/**
+ * Runs integration tests for each ES2017/ES2016 feature that
+ * is purportedly supported by the boilerplate.
+ * The difference between this and the specification tests is
+ * that while specification tests ensure that the Babel config
+ * is correct, integration tests ensure that the rest of the
+ * Javascript boilerplate code for the Lambda is working as expected
+ * and does not crash/fail to transpile.
+ *
+ * NOTE: You should remove this in your actual production code.
+ * This is only for testing the boilerplate.
+ */
+
+import run from './util/runner';
+
+it('should call Object.entries correctly', function() {
+    const event = {
+        type: 'integration-test-example',
+        data: {
+            function: 'Object.entries',
+            value: {
+                hello: 'world',
+                number: 100,
+                isTrue: true,
+            },
+        },
+    };
+
+    const result = run(event);
+
+    expect(result).toEqual([
+        ['hello', 'world'],
+        ['number', 100],
+        ['isTrue', true],
+    ]);
+});
+
+it('should call Object.values correctly', function() {
+    const event = {
+        type: 'integration-test-example',
+        data: {
+            function: 'Object.values',
+            value: {
+                hello: 'world',
+                number: 100,
+                isTrue: true,
+            },
+        },
+    };
+
+    const result = run(event);
+
+    expect(result).toEqual(['world', 100, true]);
+});
+
+it('should call Array.prototype.includes correctly', function() {
+    const event = {
+        type: 'integration-test-example',
+        data: {
+            function: 'Array.prototype.includes',
+            value: [1, 5, 70, 3, 54],
+            value2: 3,
+        },
+    };
+
+    const result = run(event);
+
+    expect(result).toEqual(true);
+});
+
+it('should call Exponentiation correctly', function() {
+    const event = {
+        type: 'integration-test-example',
+        data: {
+            function: 'Exponentiation',
+            value: 2,
+            value2: 10,
+        },
+    };
+
+    const result = run(event);
+
+    expect(result).toEqual(1024);
+});
+
+it('should call Object spread properties correctly', function() {
+    const event = {
+        type: 'integration-test-example',
+        data: {
+            function: 'Object spread properties',
+            value: {
+                hello: 'world',
+            },
+        },
+    };
+
+    const result = run(event);
+
+    expect(result).toEqual({ hello: 'world' });
+});
+
+it('should call Object rest properties correctly', function() {
+    const event = {
+        type: 'integration-test-example',
+        data: {
+            function: 'Object rest properties',
+            value: {
+                child: 'first',
+                otherChild: 'second',
+            },
+        },
+    };
+
+    const result = run(event);
+
+    expect(result).toEqual({ otherChild: 'second' });
+});

